src/components/Map/VectorTileLayer.tsx

```diff
@@ -23,12 +25,11 @@ export default function VectorTileLayer({ countries, disputedAreas }: MapProps)
     );
     MapOperations.setMapInteractionFunctionality(baseMap);
     MapOperations.synchronizeLeafletMapbox(baseMap, mapContainer, context);
-
     return () => {
       baseMap.remove();
       context.map.off('move');
     };
-  }, [context, theme]);
+  }, [context, theme, selectedMapType]);
```

Comment:
nope, this shouldn't go in here, this would cause the entire map to rerender when the map type changes, this should not be the case? Why did you need it?